azure-pipelines.yml

```diff
@@ -156,8 +156,11 @@ steps:
   # macOS
   - bash: |
       /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
-      brew install automake
+      brew install autoconf automake pkgconfig m4 libtool gettext
```

Comment:
Why do we stillneed to install autoconf? To bootstrap the older version? More generally, it looks like brew has at least `autoconf@2.13`. Does that work? That seems like a nicer solution than building our own.

Reply:
we need autoconf so that we can generate prerequisite files
```
  system_ "stack --stack-yaml hadrian/stack.yaml exec -- bash -c ./boot"
  system_ "stack --stack-yaml hadrian/stack.yaml exec -- bash -c \"./configure --enable-tarballs-autodownload\""
  ```
  we were on autocon@2.69. i haven't tried 2.13. i'll give it a go

Reply:
2.13 exhibits the same sort of errors as 2.71. i think these errors are trying to tell us there is a missing package but what? i don't want to install either of these locally because then i can get myself into a situation where i can't build ghc. fall back on the above fix for now while we work this out.

Reply:
heh. i just installed autoconf-2.71 locally (not via brew) and found that i get a lot of warnings from it but that the .boot step succeeds. so, i’m backing up now trying to understand why i felt compelled to change the azure config. i feel like i should be able to revert the azure changes and accept brew’s install of 2.71 or at least not have to install a custom autoconf.

Reply:
got this worked out. newer versions of the automake toolchain are incompatible with ghc-lib flavors <= 9.0.1. workaround by patching alocal.m4: replace `_AC_PROG_CC_C99` with `AC_PROG_CC_C99`. revert changes to `azure-pipelines.yml`.